### PR TITLE
Feature/difficulty feature

### PR DESCRIPTION
**Pull request by AMalley**
Tasks:
- [x] Documentation
- [x] Unit Tests
- [x] Typescript Typings
- [x] `npm run build:full`

### CODE REVIEW

File: src/Application.spec.js [hunk 1]
```diff
@@ -192,6 +193,40 @@ describe('Application', () => {
...
+      app.state.hitAreaScale.value = 1;
+      app.state.dragThresholdScale.value = 1;
+      app.state.health.value = 1;
+      app.state.objectCount.value = 1;
+      app.state.completionPercentage.value = 1;
+      app.state.speedScale.value = 1;
+      app.state.timersScale.value = 1;
+      app.state.inputCount.value = 1;
+
+      expect(callback.callCount).to.equal(8);
```
Comment (chipbell4) [inline src/Application.spec.js]: Love it.

File: src/README.md [hunk 2]
```diff
@@ -31,10 +31,17 @@ var myApp = new springroll.Application({
     controlSensitivity: false, // whether or not the game has adjustable control sensitivity
     buttonSize: false, // whether or not the game has adjustable button sizes
     removableLayers: false, // whether or not the game supports the removal of distracting game layers
-    hudPosition: false, // whether or not the game supports multiple HUD positions
-    difficulty: false, // whether or not the game supports an adjustable difficulty,
-    keyBinding: true, //whether or not the game supports re-mappable key bindings.
-    colorVision: true, //whether or not the game supports alternate color schemes for color blind users
+    hudPosition: false, // whether or not the game supports multiple HUD positionshitAreaScale: true,
```
Comment (902seanryan) [inline src/README.md]: Little typo here

Comment (AMalley) [reply src/README.md]: Fixed it!

File: src/README.md [hunk 3]
```diff
@@ -32,9 +32,16 @@ var myApp = new springroll.Application({
     buttonSize: false, // whether or not the game has adjustable button sizes
     removableLayers: false, // whether or not the game supports the removal of distracting game layers
     hudPosition: false, // whether or not the game supports multiple HUD positions
-    difficulty: false, // whether or not the game supports an adjustable difficulty,
-    keyBinding: true, //whether or not the game supports re-mappable key bindings.
-    colorVision: true, //whether or not the game supports alternate color schemes for color blind users
+    hitAreaScale: true, // whether or not the game supports adjustable hit areas
```
Comment (chipbell4) [inline src/README.md]: I think we should also add a section to the README to provide some examples of when these mechanics should be configurable. Some thing ala:

# Hit Area Scale
Set this feature to true if your game supports scalable hit boxes for items in your game.

☝️ Something like that.

Comment (AMalley) [reply src/README.md]: I've added documentation for these difficulty features to the read me

File: src/README.md [hunk 4]
```diff
@@ -49,6 +49,31 @@ var myApp = new springroll.Application({
...
+Enable this feature in the game to allow players to adjust the number of objects used to complete objectives. This will allow players to increase or decrease the number of items used in the game, hidden and visible.
+
+### Speed Scale
+Enable this feature if the speed of the game mechanics can increase or decrease. By slowing down or speeding up the game features the player can increase or decrease the difficulty as needed.
+
+### Completion Percentage
+Enable this feature if your game includes mechanics with a configurable completion percentage for a task. This allows the player to change the amount of times they need to interact before moving forward in the game. Example: How many swipes it takes to clear dirt off a puzzle.
+
+### Timer Scale
+Enable this feature in your game includes an adjustable timer mechanic. This allows players to give themselves more or less time to complete a challenge depending on the difficulty level they require.
```
Comment (902seanryan) [inline src/README.md]: Small typo in Timer Scale: `Enable this feature in your game...` should be `if your game includes`

File: src/README.md [hunk 5]
```diff
@@ -49,6 +49,31 @@ var myApp = new springroll.Application({
...
+
+### Input Count
+Enable this feature if your game includes a configurable mechanic that requires multiple clicks, taps or keyboard input to complete an objective. This allows players interact with the elements more or less to complete the objective depending on their difficulty requirements
```
Comment (902seanryan) [inline src/README.md]: `This allows players interact with...` should be `This allows players to interact with...`

File: src/README.md [hunk 6]
```diff
@@ -32,16 +32,48 @@ var myApp = new springroll.Application({
...
+    keyBinding: true, // whether or not the game supports re-mappable key bindings.
+    colorVision: true, // whether or not the game supports alternate color schemes for color blind users
   }
 });
 ```
 
 Note that if any of `vo`, `music`, or `sfx` are available features, `sound` will be marked as a feature implicitly.
 Also, all of these features are marked `false` by default.
 
+## Difficulty Features
```
Comment (chipbell4) [inline src/README.md]: Let's add another little section with a general overview of what the numbers mean. For instance, we should note that all values are between 0 and 1. We also might want to add implementation details about what 0 and 1 maps to in the context of that feature.

Comment (AMalley) [reply src/README.md]: Added a brief description about what the difficulty values are and what they mean

File: src/README.md (hunk 5 above)
Comment (AMalley) [reply src/README.md]: Fixed

File: src/README.md (hunk 4 above)
Comment (AMalley) [reply src/README.md]: Fixed

**Pull request merged**